### PR TITLE
python3Packages.energyflow: init at 1.3.2

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by trepetti** [comment]
Result of `nixpkgs-review pr 192781` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python310Packages.energyflow</li>
    <li>python39Packages.energyflow</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.wasserstein</li>
    <li>python39Packages.wasserstein</li>
  </ul>
</details>

**Comment by trepetti** [comment]
Result of `nixpkgs-review pr 192781` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python310Packages.energyflow</li>
    <li>python39Packages.energyflow</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.wasserstein</li>
    <li>python39Packages.wasserstein</li>
  </ul>
</details>

**Comment by trepetti** [comment]
Just wanted to verify that it didn't have anything to do with my garbage internet connection. It looks like setuptools builds now require the pytest-runner package as a native build input.

File: pkgs/development/python-modules/energyflow/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, h5py
+, numpy
```
Comment (trepetti) [inline pkgs/development/python-modules/energyflow/default.nix:5]:
```suggestion
, numpy
, pytest-runner
```

File: pkgs/development/python-modules/energyflow/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+buildPythonPackage rec {
+  pname = "EnergyFlow";
+  version = "1.3.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "e153d71a8f682956cbfa3ec6f19d5786c9bf30728f7fba2089851a16b687856e";
+  };
+
```
Comment (trepetti) [inline pkgs/development/python-modules/energyflow/default.nix:19]:
```suggestion
  nativeBuildInputs = [
    pytest-runner
  ];
  
```

**Review by trepetti — CHANGES_REQUESTED** [review]
I was able to get it to build with this change.

**Comment by veprbl** [comment]
@GrahamcOfBorg build python3Packages.energyflow

**Comment by trepetti** [comment]
> https://github.com/NixOS/nixpkgs/pull/192781/checks?check_run_id=8531991299

This second failure is the one I was getting until I added that native build input. I am not entirely sure what is going on because it seems like without it, Python tries to fetch from PyPI during the buildPhase, which is not allowed since it is sandboxed.

**Comment by veprbl** [comment]
I don't get why this issue is only seen on Linux.

**Comment by trepetti** [comment]
I think it is because sandboxing is on by default on Linux but not on MacOS. I think this can sometimes break the hermeticity of builds and still allow network access during the build phase. It is also somewhat mysterious to me, though. Maybe @jonringer or @FRidh would have more insights.

**Comment by trepetti** [comment]
> For non-Linux: Is sandbox = true set in nix.conf? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))

Could you give that a try to see if it still builds for you in that case?

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 192781` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python310Packages.energyflow</li>
    <li>python39Packages.energyflow</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.wasserstein</li>
    <li>python39Packages.wasserstein</li>
  </ul>
</details>

**Comment by veprbl** [comment]
@GrahamcOfBorg build python3Packages.energyflow

File: pkgs/development/python-modules/energyflow/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  pname = "EnergyFlow";
+  version = "1.3.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "e153d71a8f682956cbfa3ec6f19d5786c9bf30728f7fba2089851a16b687856e";
+  };
+  nativeBuildInputs = [
+    pytest-runner
+  ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/energyflow/default.nix:22]: Using pytest-runner is deprecated and should be substituted out.

File: pkgs/development/python-modules/energyflow/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  
+  propagatedBuildInputs = [
+    h5py
+    numpy
+    six
+    wasserstein
+  ];
+
+  doCheck = false; # no tests in the PyPi tarball
```
Comment (mweinelt) [inline pkgs/development/python-modules/energyflow/default.nix:31]: Then let's fetch the source from Git. https://github.com/pkomiske/EnergyFlow

File: pkgs/development/python-modules/wasserstein/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    sha256 = "dc95415bb91a3b6d04d06f11878fcfab56468311a4991da2488600f6abc7fd5b";
+  };
+
+  propagatedBuildInputs = [
+    numpy
+    llvmPackages.openmp
+    wurlitzer
+  ];
+
+  doCheck = false; # no tests in the PyPi tarball
```
Comment (mweinelt) [inline pkgs/development/python-modules/wasserstein/default.nix:25]: Let's fetch the source from GitHub here as well.

File: pkgs/development/python-modules/wasserstein/default.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  version = "1.1.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "dc95415bb91a3b6d04d06f11878fcfab56468311a4991da2488600f6abc7fd5b";
+  };
+
+  propagatedBuildInputs = [
+    numpy
+    llvmPackages.openmp
```
Comment (mweinelt) [inline pkgs/development/python-modules/wasserstein/default.nix:21]: Probably better kept in `buildInputs`.

**Review by mweinelt — CHANGES_REQUESTED** [review]

**Comment by veprbl** [comment]
@GrahamcOfBorg build python3Packages.energyflow

File: pkgs/development/python-modules/energyflow/default.nix [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  '';
+
+  propagatedBuildInputs = [
+    h5py
+    numpy
+    six
+    wasserstein
+  ];
+
+  doCheck = true;
```
Comment (mweinelt) [inline pkgs/development/python-modules/energyflow/default.nix:35]: For python builds this is implicitly true.

File: pkgs/development/python-modules/energyflow/default.nix [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  checkInputs = [
+    pytestCheckHook
+  ];
+  pytestFlagsArray = [
+    "energyflow/tests"
+  ];
+  preCheck = ''
+    rm energyflow/tests/test_archs.py # requires tensorflow
+    rm energyflow/tests/test_emd.py # requires "ot"
+  '';
```
Comment (mweinelt) [inline pkgs/development/python-modules/energyflow/default.nix:45]: We do have tensorflow, but I assume you don't want to create a dependency onto that?

```suggestion
  disabledTestPaths = ['
    "energyflow/tests/test_archs.py" # requires tensorflow
    "energyflow/tests/test_emd.py" # requires "ot
  ];
```

**Review by mweinelt — CHANGES_REQUESTED** [review]
For both packages:

- Remove `doCheck = true`, it is implicitly the case for python builds.
- Use `disabledTestPaths`, which is provided by `pytestCheckHook`

File: pkgs/development/python-modules/energyflow/default.nix (hunk 8 above)
Comment (veprbl) [reply pkgs/development/python-modules/energyflow/default.nix]: Seems like people are not so careful when it comes to tensorflow maintenance in nixpkgs. I'd skip on this dependency for now.

**Pull request merged**